File: Yang params model/dengue_model/muL_a.py
from __future__ import annotations

import numpy as np

from ._utils import return_like_input
from .temp import temp


# MATLAB usa polyval(fliplr(coefficients), x), asi que aqui los almacenamos
# ya invertidos para reproducir exactamente ese comportamiento.
COEFFICIENTS = np.array([6.794e-06, -6.778e-04, 2.457e-02, -3.797e-01, 2.130], dtype=float)


def muL_a(day):
    x_values = temp(day, 0)
    result = np.polyval(COEFFICIENTS, x_values)
    return return_like_input(day, result)

File: Yang params model/matlab_original/muL_a.m
%YANG 2011
function muL_a = muL_a(day)
    global P 
    coefficients = [2.130, -3.797*10^(-1), 2.457*10^(-2), -6.778*10^(-4), 6.794*10^(-6)];
    x_values=temp(day, 0);
    muL_a = polyval(fliplr(coefficients), x_values);
    %plot(linspace(10,40,40), polyval(fliplr(coefficients),
    %(linspace(10,40,40))))  to plot temp dependent params
end
